ocaml/typing/predef.ml

```diff
@@ -472,6 +489,8 @@ let add_small_number_extension_types add_type env =
   let add_type = mk_add_type add_type in
```

Comment:
(This is as close as I can comment) @TheNumbat: Should the SIMD types mode-cross? I would assume that they are well-behaved immutable data.

Reply:
Yes that sounds like an interesting question. If this turns out to be more complex than a few lines of code, I suggest we leave this as future work. (This PR is needed to unblock some other projects).

Reply:
Values of SIMD types - are they pointers to a region in memory? Is that memory managed by OCaml?

Reply:
Yeah, they should be immutable data. You can treat them the same as floats; the unboxed versions live in a single register and the boxed versions are a pointer to an immutable two-word custom block.

Reply:
(Same goes for `float32`.)

Reply:
Thanks - added.